src/OrbitGl/GraphTrack.h

```diff
@@ -31,10 +31,21 @@ class GraphTrack : public Track {
                         PickingMode picking_mode, float z_offset = 0) override;
   [[nodiscard]] float GetHeight() const override;
   void AddValue(double value, uint64_t time);
-  [[nodiscard]] std::optional<std::pair<uint64_t, double> > GetPreviousValueAndTime(
+  [[nodiscard]] std::optional<std::pair<uint64_t, double>> GetPreviousValueAndTime(
       uint64_t time) const;
   [[nodiscard]] bool IsEmpty() const override { return values_.empty(); }
 
+  void SetWarningThreshold(const std::optional<std::pair<std::string, double>>& warning_threshold) {
```

Comment:
I don't like to have this type in the set function.

What do you think about:
- Have a ClearWarningThreshold function to set this to false.
- Have also a SetWarningThreshold which takes an string and also a double and assumes that it is true?

And also don't ask for them in the constructors. 

Also maybe for a future PR, I would like to have it only in a MemoryTrack, and not in all the GraphTracks. So, MemoryTrack inherit from GraphTrack and also having this SetWarningThreshold and ClearWarningThreshold methods.

Reply:
Make sense. As we discussed offline, we may have a specific implementation for the memory tracks. This would be helpful especially when we do a type based filtering to hide and show tracks. Because memory timers, value tracking timers and introspection timers can all be GraphTracks, so with `Track::Type` we cannot differentiate the purpose of different tracks. So I agree that we can let the MemoryTrack to be inherit from the GraphTrack.

Reply:
Sure! And what do you think about having this SetWarningThreshold (string , double) instead of the optional <pair> (what I wrote in the top part)?

Reply:
Sounds good! So we can keep the type of `warning_threshold_` but set it with string and double instead of optional. I will change accordingly : )